cf-oscore/src/main/java/org/eclipse/californium/oscore/Encryptor.java

```diff
@@ -209,8 +214,15 @@ protected static Encrypt0Message prepareCOSEStructure(byte[] confidential, byte[
 			}
 		} else {
 			bRes.write(firstByte);
-			return bRes.toByteArray();
 		}
-		return bRes.toByteArray();
+		
+		//If the OSCORE option is length 1 and 0x00, it can be empty
+		//See https://tools.ietf.org/html/draft-ietf-core-object-security-16#page-33 point 4
+		byte[] optionBytes = bRes.toByteArray();
+		if(optionBytes.length == 1 && optionBytes[0] == 0x00) {
+			return Bytes.EMPTY;
```

Comment:
I guess, this change causes now
`OSCoreTest.testEndcodingObjectSecurityValueCompression` to fail when
`assertArrays(objectSecurityResponse, new byte[] { 0x00 });`

It may be 
`assertArrayEquals(objectSecurityResponse, Bytes.EMPTY);`

(I would prefer to replace the own implementation of `assertArrays` by the one of `org.junit.Assert`. Or do you have a reason for your own implementation?)

Reply:
Yes, I changed the comparison to Bytes.EMPTY instead since that should now be the correct value. I also removed the assertArrays method and instead use the one from org.junit.Assert as you recommend. Done in commit 3bf97445eb5c2bb814e0f2e5b10787b3b3e1b31f